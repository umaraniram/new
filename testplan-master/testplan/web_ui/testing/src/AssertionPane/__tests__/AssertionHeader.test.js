import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";

import AssertionHeader from '../AssertionHeader';

function defaultProps() {
  return {
    assertion: {
      "category":"DEFAULT",
      "machine_time":"2019-02-12T17:41:42.795536+00:00",
      "description":null,
      "line_no":25,
      "label":"==",
      "second":"foo",
      "meta_type":"assertion",
      "passed":true,
      "type":"Equal",
      "utc_time":"2019-02-12T17:41:42.795530+00:00",
      "first":"foo"
    },
    index: 0,
    onClick: jest.fn()
  };
}


describe('AssertionHeader', () => {
  let props;
  let shallowComponent;

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
    props = defaultProps();
    shallowComponent = undefined;
  });

  it('shallow renders the correct HTML structure', () => {
    shallowComponent = shallow(<AssertionHeader {...props}/>);
    expect(shallowComponent).toMatchSnapshot();
  });
});