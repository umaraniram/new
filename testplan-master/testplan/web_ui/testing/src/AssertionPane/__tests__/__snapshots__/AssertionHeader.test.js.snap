// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssertionHeader shallow renders the correct HTML structure 1`] = `
<CardHeader
  className="cardHeader_1ggz856-o_O-cardHeaderColorPassed_1nxwz58"
  onClick={[MockFunction]}
  tag="div"
>
  <span>
    <strong />
  </span>
  <span>
    (
    Equal
    )
  </span>
  <span
    className="cardHeaderAlignRight_107ja4p"
  />
</CardHeader>
`;
